tests/components/template/test_binary_sensor.py

```diff
@@ -15,207 +14,198 @@
-        config = {
-            "binary_sensor": {
-                "platform": "template",
-                "sensors": {
-                    "test": {
-                        "friendly_name": "virtual thingy",
-                        "value_template": "{{ foo }}",
-                        "device_class": "motion",
-                    }
-                },
-            }
+async def test_setup(hass):
+    """Test the setup."""
+    config = {
+        "binary_sensor": {
+            "platform": "template",
+            "sensors": {
+                "test": {
+                    "friendly_name": "virtual thingy",
+                    "value_template": "{{ foo }}",
+                    "device_class": "motion",
+                }
+            },
         }
-        with assert_setup_component(1):
-            assert setup.setup_component(self.hass, "binary_sensor", config)
-
-    def test_setup_no_sensors(self):
-        """Test setup with no sensors."""
-        with assert_setup_component(0):
-            assert setup.setup_component(
-                self.hass, "binary_sensor", {"binary_sensor": {"platform": "template"}}
-            )
-
-    def test_setup_invalid_device(self):
-        """Test the setup with invalid devices."""
-        with assert_setup_component(0):
-            assert setup.setup_component(
-                self.hass,
-                "binary_sensor",
-                {"binary_sensor": {"platform": "template", "sensors": {"foo bar": {}}}},
-            )
-
-    def test_setup_invalid_device_class(self):
-        """Test setup with invalid sensor class."""
-        with assert_setup_component(0):
-            assert setup.setup_component(
-                self.hass,
-                "binary_sensor",
-                {
-                    "binary_sensor": {
-                        "platform": "template",
-                        "sensors": {
-                            "test": {
-                                "value_template": "{{ foo }}",
-                                "device_class": "foobarnotreal",
-                            }
-                        },
-                    }
-                },
-            )
-
-    def test_setup_invalid_missing_template(self):
-        """Test setup with invalid and missing template."""
-        with assert_setup_component(0):
-            assert setup.setup_component(
-                self.hass,
-                "binary_sensor",
-                {
-                    "binary_sensor": {
-                        "platform": "template",
-                        "sensors": {"test": {"device_class": "motion"}},
-                    }
-                },
-            )
+    }
+    with assert_setup_component(1):
+        assert await setup.async_setup_component(hass, binary_sensor.DOMAIN, config)
+
+
+async def test_setup_no_sensors(hass):
+    """Test setup with no sensors."""
+    with assert_setup_component(0):
+        assert await setup.async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6319846)